cmd/kueuectl/app/list/list_pods_printer.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package list
+
+import (
+	"errors"
+	"io"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/cli-runtime/pkg/printers"
+)
+
+type listPodPrinter struct {
+	printOptions printers.PrintOptions
+}
+
+var _ printers.ResourcePrinter = (*listPodPrinter)(nil)
+
+func (p *listPodPrinter) PrintObj(obj runtime.Object, out io.Writer) error {
+	printer := printers.NewTablePrinter(p.printOptions)
+
+	list, ok := obj.(*corev1.PodList)
+	if !ok {
+		return errors.New("invalid object type")
+	}
+
+	table := &metav1.Table{
+		ColumnDefinitions: []metav1.TableColumnDefinition{
+			{Name: "Name", Type: "string", Format: "name"},
```

Comment:
Sure, looking if I can reuse some helper functions from Kubectl printers itself.

Reply:
Do you want me to modify our printer to use [this](https://github.com/kubernetes/kubernetes/blob/master/pkg/printers/internalversion/printers.go#L861) func. I feel like we might need to show whateever `kubectl get po` shows or should we just print only `Ready` & `Restarts`?

Reply:
Thanks, working on it. For printer can we just use [table_printer](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/get/table_printer.go)?

Reply:
Yes, sure.

Reply:
okay I was able to get the logic working in local. But for unit testing it seems, I need to replicate like [here](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/get/get_test.go#L2794), for labelSelector, fieldSelector queries etc., ? Also, handling different request for example we have `*jobs` requests as well and adding to that now we have `pods` too.

For example:
```go
	tf.UnstructuredClient = &restfake.RESTClient{
		NegotiatedSerializer: resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer,
		Client: restfake.CreateHTTPClient(func(request *http.Request) (*http.Response, error) {
			switch request.URL.Path {
			case "/namespaces/default/jobs":
				return &http.Response{
					StatusCode: http.StatusOK,
					Header:     cmdtesting.DefaultHeader(),
					Body:       io.NopCloser(bytes.NewReader([]byte(runtime.EncodeOrDie(codec, tc.job[0])))),
				}, nil
			case "/namespaces/default/pods":
				return &http.Response{
					StatusCode: http.StatusOK,
					Header:     cmdtesting.DefaultHeader(),
					Body:       podTableObjBody(codec, podList.Items...),
				}, nil
			default:
				t.Fatalf("request URL: %#v, and request: %#v", request.URL, request)
				return nil, nil
			}
		}),
```
^^ this is just a sample one. Just checking to see if I'm moving in the right direction.

Reply:
Yes, to test it we need to mock `UnstructuredClient`.

Reply:
But maybe it's better to move this switch-case logic to test case?

Reply:
Yeah was thinking the same. Will look into it.

Reply:
> Thanks, working on it. For printer can we just use [table_printer](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/get/table_printer.go)?

Also need to check that `-o` flag working fine.

Reply:
Yeah `-o=wide` is working fine, added a test for the same. I just pushed an initial version https://github.com/kubernetes-sigs/kueue/pull/2280/commits/ded5836dd228e8d5a225796bf695a327f7471903 might need some refactoring.

Also, in tests we are not handling the labelSelection (based on jobs) anywhere, since the actual server resp needs to mocked, I'm not sure what's the cleaner way to do here. lmk your thoughts.

Reply:
> Yeah -o=wide is working fine, added a test for the same. 

But also I mean `-o yaml` and `-o json`.

Reply:
Label selector and field selector hard to test in unit tests. You can use integration test to check it. https://github.com/kubernetes-sigs/kueue/blob/ded5836dd228e8d5a225796bf695a327f7471903/test/integration/kueuectl/list_test.go.

Reply:
Overall looks good to me.